sdk/sdk.go

```diff
@@ -9,29 +9,40 @@ import (
 	ringtypes "github.com/athanorlabs/go-dleq/types"
 	"github.com/noot/ring-go"
 	"github.com/pokt-network/poktroll/pkg/crypto/rings"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
 	servicetypes "github.com/pokt-network/poktroll/x/service/types"
 
 	"github.com/pokt-network/shannon-sdk/types"
 )
 
+// ApplicationLister returns all the applications or a single application with the specified address.
+//
+//	It is used by the SDK to perform functions related to applications, e.g. listing applications delegating to a gateway address.
+//
+// DISCUSS: it may be possible to remove the need for this through providing helper functions and/or methods on the application struct.
```

Comment:
```suggestion
// TODO_DISCUSS: it may be possible to remove the need for this through providing helper functions and/or methods on the application struct.
```

Reply:
thank you, fixed.